EngineLayer/Proteomics/PeptideWithPossibleModifications.cs

```diff
@@ -25,6 +25,14 @@ internal PeptideWithPossibleModifications(int oneBasedStartResidueNumberInProtei
             this.NumKnownPossibleLocMods = this.Protein.OneBasedPossibleLocalizedModifications.Count(kvp => kvp.Key >= OneBasedStartResidueInProtein && kvp.Key <= OneBasedEndResidueInProtein);
         }
 
+        internal PeptideWithPossibleModifications(int oneBasedStartResidueNumberInProtein, int oneBasedEndResidueNumberInProtein, Protein parentProtein, int missedCleavages, string peptideDescription, Dictionary<int, ModificationWithMass> thisDictionaryOfFixedMods)
```

Comment:
Instead of this constructor, add a CutPiece method to PeptideWithPossibleModifications, that creates a new shorter PeptideWithPossibleModifications object by means of calling the original constructor, with needed parameters.